sdk/keyvault/azure-security-keyvault-certificates/src/main/java/com/azure/security/keyvault/certificates/CertificateClientBuilder.java

```diff
@@ -294,12 +359,18 @@ public CertificateClientBuilder serviceVersion(CertificateServiceVersion version
      * The default retry policy will be used in the pipeline, if not provided.
      *
      * @param retryPolicy user's retry policy applied to each request.
-     * @return The updated CertificateClientBuilder object.
-     * @throws NullPointerException if the specified {@code retryPolicy} is null.
+     *
+     * @return The updated {@link CertificateClientBuilder} object.
+     *
+     * @throws NullPointerException If the specified {@code retryPolicy} is null.
      */
     public CertificateClientBuilder retryPolicy(RetryPolicy retryPolicy) {
-        Objects.requireNonNull(retryPolicy, "The retry policy cannot be bull");
+        if (retryPolicy == null) {
```

Comment:
Should null be allowed here and a default be used if it is set to null?

Reply:
We can do that, sure.